pkg/cmd/workflow/shared/shared.go

```diff
@@ -0,0 +1,157 @@
+package shared
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/prompt"
+)
+
+const (
+	Active           WorkflowState = "active"
+	DisabledManually WorkflowState = "disabled_manually"
+)
+
+type WorkflowState string
+
+type Workflow struct {
+	Name  string
+	ID    int
+	Path  string
+	State WorkflowState
+}
+
+type WorkflowsPayload struct {
+	Workflows []Workflow
+}
+
+func (w *Workflow) Disabled() bool {
+	return w.State != Active
+}
+
+func (w *Workflow) Base() string {
+	return path.Base(w.Path)
+}
+
+func GetWorkflows(client *api.Client, repo ghrepo.Interface, limit int) ([]Workflow, error) {
+	perPage := limit
+	page := 1
+	if limit > 100 || limit == 0 {
+		perPage = 100
+	}
+
+	workflows := []Workflow{}
+
+	for {
+		if limit > 0 && len(workflows) == limit {
+			break
+		}
+		var result WorkflowsPayload
+
+		path := fmt.Sprintf("repos/%s/actions/workflows?per_page=%d&page=%d", ghrepo.FullName(repo), perPage, page)
+
+		err := client.REST(repo.RepoHost(), "GET", path, nil, &result)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, workflow := range result.Workflows {
+			workflows = append(workflows, workflow)
+			if limit > 0 && len(workflows) == limit {
+				break
+			}
+		}
+
+		if len(result.Workflows) < perPage {
+			break
+		}
+
+		page++
+	}
+
+	return workflows, nil
+}
+
+func SelectWorkflow(workflows []Workflow, promptMsg string, states []WorkflowState) (*Workflow, error) {
```

Comment:
From what I can tell the calls to this function only use `active` as the `states` argument, do you foresee this being used with other states in the future?